### DISCUSSION TITLE
gyp ERR! find VS - does not match this Visual Studio Command Prompt

### DISCUSSION THREAD

**Discussion by keerthikumarb** [eclipse-theia/theia#14492] [Q&A]
I am trying to build "Coffee-Editor" example.
Though the backend is built properly, I face issues building the frontend code.

$ yarn client --ignore-engines && yarn client development
$ yarn -s --cwd client --ignore-engines

Below is the error message:

warning Error running install script for optional dependency: "C:\\Projects\\coffee-editor\\client\\node_modules\\cpu-features: Command failed.
Exit code: 1
Command: node buildcheck.js > buildcheck.gypi && node-gyp rebuild
Arguments:
Directory: C:\\Projects\\coffee-editor\\client\\node_modules\\cpu-features
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@9.4.1
gyp info using node@20.18.0 | win32 | x64
gyp info find Python using Python version 3.13.0 found at \"C:\\Users\\MyPC\\AppData\\Local\\Programs\\Python\\Python313\\python.exe\"
gyp ERR! find VS
gyp ERR! find VS msvs_version not set from command line or npm config
gyp ERR! find VS running in VS Command Prompt, installation path is:
gyp ERR! find VS \"C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\BuildTools\\Common7\"
gyp ERR! find VS - will only use this version
gyp ERR! find VS checking VS2017 (15.9.35431.46) found at:
gyp ERR! find VS \"C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\BuildTools\"
gyp ERR! find VS - found \"Visual Studio C++ core features\"
gyp ERR! find VS - found VC++ toolset: v141
gyp ERR! find VS - found Windows SDK: 10.0.17763.0
gyp ERR! find VS - does not match this Visual Studio Command Prompt
gyp ERR! find VS checking VS2017 (15.9.35431.46) found at:
gyp ERR! find VS \"C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Professional\"
gyp ERR! find VS - found \"Visual Studio C++ core features\"
gyp ERR! find VS - found VC++ toolset: v141
gyp ERR! find VS - found Windows SDK: 10.0.17763.0
**gyp ERR! find VS - does not match this Visual Studio Command Prompt**
gyp ERR! find VS could not find a version of Visual Studio 2017 or newer to use
gyp ERR! find VS not looking for VS2015 as it is only supported up to Node.js 18
gyp ERR! find VS not looking for VS2013 as it is only supported up to Node.js 8
gyp ERR! find VS
gyp ERR! find VS **************************************************************
gyp ERR! find VS You need to install the latest version of Visual Studio
gyp ERR! find VS including the \"Desktop development with C++\" workload.
gyp ERR! find VS For more information consult the documentation at:
gyp ERR! find VS https://github.com/nodejs/node-gyp#on-windows
gyp ERR! find VS **************************************************************
gyp ERR! find VS
gyp ERR! configure error
gyp ERR! stack Error: Could not find any Visual Studio installation to use
gyp ERR! stack     at VisualStudioFinder.fail (C:\\Projects\\coffee-editor\\client\\node_modules\\node-gyp\\lib\\find-visualstudio.js:122:47)
gyp ERR! stack     at C:\\Projects\\coffee-editor\\client\\node_modules\\node-gyp\\lib\\find-visualstudio.js:75:16
gyp ERR! stack     at VisualStudioFinder.findVisualStudio2013 (C:\\Projects\\coffee-editor\\client\\node_modules\\node-gyp\\lib\\find-visualstudio.js:380:14)
gyp ERR! stack     at C:\\Projects\\coffee-editor\\client\\node_modules\\node-gyp\\lib\\find-visualstudio.js:71:14
gyp ERR! stack     at VisualStudioFinder.findVisualStudio2015 (C:\\Projects\\coffee-editor\\client\\node_modules\\node-gyp\\lib\\find-visualstudio.js:364:14)
gyp ERR! stack     at C:\\Projects\\coffee-editor\\client\\node_modules\\node-gyp\\lib\\find-visualstudio.js:67:12
gyp ERR! stack     at VisualStudioFinder.parseData (C:\\Projects\\coffee-editor\\client\\node_modules\\node-gyp\\lib\\find-visualstudio.js:237:5)
gyp ERR! stack     at C:\\Projects\\coffee-editor\\client\\node_modules\\node-gyp\\lib\\find-visualstudio.js:143:14
gyp ERR! stack     at ChildProcess.exithandler (node:child_process:414:7)
gyp ERR! stack     at ChildProcess.emit (node:events:519:28)
gyp ERR! stack     at maybeClose (node:internal/child_process:1105:16)
gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:305:5)
gyp ERR! System Windows_NT 10.0.19045
gyp ERR! command \"C:\\\\Program Files\\\\nodejs\\\\node.exe\" \"C:\\\\Projects\\\\coffee-editor\\\\client\\\\node_modules\\\\node-gyp\\\\bin\\\\node-gyp.js\" \"rebuild\"
gyp ERR! cwd C:\\Projects\\coffee-editor\\client\\node_modules\\cpu-features
gyp ERR! node -v v20.18.0
gyp ERR! node-gyp -v v9.4.1
gyp ERR! not ok"

**Comment by msujew** [comment]
Hey @keerthikumarb,

It seems like the native build tools required by node-gyp weren't correctly installed in your machine. Can you try uninstalling node and reinstalling it via the msi installer? Make sure to select the option to install the native build tools. Afterwards, rerunning the build process should hopefully work as expected.

By the way, cpufeatures is a dependency of the backend, which is failing to build, not the frontend.

**Reply by keerthikumarb** [reply]
Hello Mark,

Thank you for the reply.
I have tried the suggestion mentioned above. Sadly, the issue persists.
Now I tried with other examples and the result is the same.

error C:\Projects\my-extension\node_modules\@theia\ffmpeg: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: C:\Projects\my-extension\node_modules\@theia\ffmpeg
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@9.4.1
gyp info using node@18.17.0 | win32 | x64
gyp info find Python using Python version 3.13.0 found at "C:\Users\MyPC\AppData\Local\Programs\Python\Python313\python.exe"
gyp http GET https://nodejs.org/download/release/v18.17.0/node-v18.17.0-headers.tar.gz
gyp http 200 https://nodejs.org/download/release/v18.17.0/node-v18.17.0-headers.tar.gz
gyp http GET https://nodejs.org/download/release/v18.17.0/SHASUMS256.txt
gyp http GET https://nodejs.org/download/release/v18.17.0/win-x64/node.lib
gyp http 200 https://nodejs.org/download/release/v18.17.0/SHASUMS256.txt
gyp http 200 https://nodejs.org/download/release/v18.17.0/win-x64/node.lib
gyp ERR! find VS 
gyp ERR! find VS msvs_version not set from command line or npm config
gyp ERR! find VS running in VS Command Prompt, installation path is:
gyp ERR! find VS "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7"
gyp ERR! find VS - will only use this version
gyp ERR! find VS checking VS2019 (16.11.35425.106) found at:
gyp ERR! find VS "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools"
gyp ERR! find VS - found "Visual Studio C++ core features"
gyp ERR! find VS - found VC++ toolset: v142
gyp ERR! find VS - found Windows SDK: 10.0.19041.0
gyp ERR! find VS - does not match this Visual Studio Command Prompt
gyp ERR! find VS checking VS2017 (15.9.35431.46) found at:
gyp ERR! find VS "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools"
gyp ERR! find VS - found "Visual Studio C++ core features"
gyp ERR! find VS - found VC++ toolset: v141
gyp ERR! find VS - found Windows SDK: 10.0.17763.0
gyp ERR! find VS - does not match this Visual Studio Command Prompt
gyp ERR! find VS checking VS2017 (15.9.35431.46) found at:
gyp ERR! find VS "C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional"
gyp ERR! find VS - found "Visual Studio C++ core features"
gyp ERR! find VS - found VC++ toolset: v141
gyp ERR! find VS - found Windows SDK: 10.0.17763.0
gyp ERR! find VS - does not match this Visual Studio Command Prompt
gyp ERR! find VS could not find a version of Visual Studio 2017 or newer to use
gyp ERR! find VS looking for Visual Studio 2015
gyp ERR! find VS - not found
gyp ERR! find VS not looking for VS2013 as it is only supported up to Node.js 8
gyp ERR! find VS
gyp ERR! find VS **************************************************************
gyp ERR! find VS You need to install the latest version of Visual Studio
gyp ERR! find VS including the "Desktop development with C++" workload.
gyp ERR! find VS For more information consult the documentation at:
gyp ERR! find VS https://github.com/nodejs/node-gyp#on-windows
gyp ERR! find VS **************************************************************
gyp ERR! find VS
gyp ERR! configure error
gyp ERR! stack Error: Could not find any Visual Studio installation to use
gyp ERR! stack     at VisualStudioFinder.fail (C:\Projects\my-extension\node_modules\node-gyp\lib\find-visualstudio.js:122:47)
gyp ERR! stack     at C:\Projects\my-extension\node_modules\node-gyp\lib\find-visualstudio.js:75:16
gyp ERR! stack     at VisualStudioFinder.findVisualStudio2013 (C:\Projects\my-extension\node_modules\node-gyp\lib\find-visualstudio.js:380:14)
gyp ERR! stack     at C:\Projects\my-extension\node_modules\node-gyp\lib\find-visualstudio.js:71:14
gyp ERR! stack     at C:\Projects\my-extension\node_modules\node-gyp\lib\find-visualstudio.js:401:16
gyp ERR! stack     at C:\Projects\my-extension\node_modules\node-gyp\lib\util.js:54:7
gyp ERR! stack     at C:\Projects\my-extension\node_modules\node-gyp\lib\util.js:33:16
gyp ERR! stack     at ChildProcess.exithandler (node:child_process:427:5)
gyp ERR! stack     at ChildProcess.emit (node:events:514:28)
gyp ERR! stack     at maybeClose (node:internal/child_process:1091:16)
gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:302:5)
gyp ERR! System Windows_NT 10.0.19045
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Projects\\my-extension\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Projects\my-extension\node_modules\@theia\ffmpeg

Is there any other way to deal with this problem?
